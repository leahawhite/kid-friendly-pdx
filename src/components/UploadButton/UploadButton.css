.icon-none {
  display: none;
}

.upload-btn {
  display: flex;
  align-items: center;
  justify-content: space-around;
  flex-wrap: wrap;
  margin: 20px;
  color: #00b8e6;
}

.button {
  cursor: pointer;
}

input[type=file] { 
  visibility: hidden;
  position: absolute;
} 

.upload-btn-text {
  margin-left: 10px;
}