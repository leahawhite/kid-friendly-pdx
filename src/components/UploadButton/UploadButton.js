import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default props => 
  <div className='buttons fadein'>
    <div className='button'>
      <label htmlFor='multi'>
        <FontAwesomeIcon icon='camera' size='10x' />
      </label>
      <input type='file' id='multi' onChange={props.onChange} multiple />
    </div>
  </div>