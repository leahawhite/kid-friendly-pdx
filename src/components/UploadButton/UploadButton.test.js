import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UploadButton from './UploadButton';

describe('UploadButton component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UploadButton />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<UploadButton />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})