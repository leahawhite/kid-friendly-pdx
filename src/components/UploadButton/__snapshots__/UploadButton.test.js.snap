// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton component renders UI as expected 1`] = `
<div
  className="buttons fadein"
>
  <div
    className="button"
  >
    <label
      className="upload-btn"
      htmlFor="multi"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="camera"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="10x"
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <span>
        Browse Files
      </span>
    </label>
    <input
      id="multi"
      multiple={true}
      type="file"
    />
  </div>
</div>
`;
